app/src/main/java/uw/hcrlab/kubi/lesson/PromptData.java

```diff
@@ -47,27 +47,76 @@ public String toString() {
                 this.type, this.PromptText, wordString, optionsString, imagesString);
     }
 
+    public static class Hint {
+        // will extend so this can be an "explain" or "conjugate" button when applicable
+        public String text;
+
+        public Hint setText(String text) {
+            this.text = text;
+            return this;
+        }
+
+        public String toString() {
+            return String.format(Locale.US, "Hint {%s}", this.text);
+        }
+    }
+
+    public static class HintCollection {
+        public ArrayList<Hint> details;
+
+        public HintCollection() {
+            this.details = new ArrayList<>();
+        }
+
+        public String toString() {
+            String detailString = "";
+            for (Hint detail: this.details) {
+                detailString += detail.toString() + ", ";
+            }
+            return String.format(Locale.US, "HintCollection {details=%s}", detailString);
+        }
+
+        public boolean isEmpty() {
+            return this.details.size() < 1;
+        }
+    }
+
```

Comment:
It's convenient for them to be in one file, but I don't think it makes sense for them to be sub-types like they currently are. For instance, this current structure gives a Word object access to private static members of the PromptData class, but there is no reason why a Word object should be able to reach into the PromptData class.